src/libstore/s3_binary_cache_store.go

```diff
@@ -0,0 +1,101 @@
+package libstore
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"net/url"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"log"
+)
+
+// S3BinaryCacheStore ...
+type S3Store struct {
+	Profile string // The name of the AWS configuration profile to use.
+	Region string // "aws-region"
+	Scheme string //  "The scheme to use for S3 requests, https by default."
+	// narinfoCompression string // "compression method for .narinfo files"
+	// multipartUpload bool
+	// bufferSize uint64
+ // do we need to put endpoint in here
+ // anything that is needed in minio, no need here. Check.
+	bucketName string
+  client *minio.Client
+}
+
+// SDK should be in environment
+func (s *S3Store) getUri() string {
+  return "s3://" + s.bucketName
+}
+
+func (s *S3Store) FileExists(ctx context.Context, path string) (bool, error) {
+
+	return true
```

Comment:
This is not complete, it should only return true if `s.client.GetObject` doesn't return a 404.